ld/app.ld

```diff
@@ -0,0 +1,99 @@
+/*
+ * app.ld -- app link
+ */
+
+MEMORY {
+	ROM   (RX)  : ORIGIN = 0x00000000, LENGTH = 16K
+	BLFLASH (RX)  : ORIGIN = 0x10000000, LENGTH = 16K
+	APPFLASH (RX)  : ORIGIN = ORIGIN(BLFLASH) + LENGTH(BLFLASH), LENGTH = 2048K - LENGTH(BLFLASH)
+	SRAM  (RWX) : ORIGIN = 0x20000000, LENGTH = 256K
+	SRAM4 (RWX) : ORIGIN = 0x20040000, LENGTH = 4K
+	SRAM5 (RWX) : ORIGIN = 0x20041000, LENGTH = 4K
+}
+
+HIDDEN(vector_align = 256);
+HIDDEN(word_align = 4);
+
+ENTRY(_start)
+
+SECTIONS {
+	.app_header (READONLY) : ALIGN(word_align) {
```

Comment:
You should add an assertion to make sure that the app header is where you expect it to be and the correct size.